### PR TITLE
Add Log Tail Wrapper

### PR DESCRIPTION
**Pull request by postables**
I needed this functionality to be able to tail ipfs logs for another project and figured it would probably be useful to others.

I tried using the built in request builder, however it appears that it just continually waits for the call to finish, and given that the `log/tail` call streams responses, so it just hangs.

### CODE REVIEW

**Comment by Stebalien** [comment]
`Decode` decodes and then drains the rest. We'll want something like the pubsub decoder. However, see https://github.com/ipfs/go-ipfs-api/pull/180.

**Comment by postables** [comment]
Ah okay, I'll go back to the drawing board and refactor :ok_hand:

**Comment by postables** [comment]
Ok I have made the change and tested locally and it works. Really unsure why the ci test are failing, however I don't think its related to this code, as the same problem is effecting your ci build for #180

File: logger.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+// GetLogs is used to retrieve a parsable logger object
+func (s *Shell) GetLogs(ctx context.Context) (Logger, error) {
+
+	resp, err := s.Request("log/tail").Send(ctx)
+	if err != nil {
+		return Logger{}, nil
+	}
+	if resp.Error != nil {
+		resp.Close()
```
Comment (Stebalien) [inline logger.go]: Let's call `resp.Output.Close()` resp.Close()` drains the stream (needed for other requests where closing early will abort the request.

File: logger_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package shell
+
+import (
+	"context"
+	"testing"
+)
+
+func Test_Logger(t *testing.T) {
```
Comment (Stebalien) [inline logger_test.go]: Nit: `TestLogger`

File: logger.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+func (l Logger) Close() error {
+	return l.resp.Close()
+}
+
+// GetLogs is used to retrieve a parsable logger object
+func (s *Shell) GetLogs(ctx context.Context) (Logger, error) {
+
+	resp, err := s.Request("log/tail").Send(ctx)
+	if err != nil {
+		return Logger{}, nil
```
Comment (Stebalien) [inline logger.go]: This is the bug, it should return `err`.

File: logger.go (hunk 1 above)
Comment (postables) [reply logger.go]: Ah good point.

File: logger.go (hunk 3 above)
Comment (postables) [reply logger.go]: Ah yes good catch!

**Comment by postables** [comment]
Ok I believe I've made the fixes. If wanted I can also port this logging stuff over to `go-ipfs-http-client` later this weekend.

**Comment by Stebalien** [comment]
> If wanted I can also port this logging stuff over to go-ipfs-http-client later this weekend.

That's mostly for exposing a stable, "core" API so we'll have to think through this a bit. IMO, we should consider simply exposing raw requests and leaving it at that (but I'll let @magik6k make the hard decisions here :)).

**Comment by magik6k** [comment]
> If wanted I can also port this logging stuff over to go-ipfs-http-client later this weekend.

Logging is rather go-ipfs specific, and afaik there were some plans to rewrite go-log, so the interface might not be too stable.

I want to rewrite most of go-ipfs-api on top of CoreAPI (with things like logging using raw requests if on top of go-ipfs-http-client, erroring if not). I should be able to PR something this week.

Options here are:
* Wait for go-ipfs-api backed by CoreAPI, use `NewCoreShell(httpCoreApi).GetLogs()`
* Add 2 interfaces to interface-go-ipfs-core - one with logging related functions, second with `Logging() LoggingAPI`, then implement it in `http-client`, and optionally `go-ipfs`

**Pull request merged**